File: Back/src/main/java/com/showbook/back/entity/Member.java
package com.showbook.back.entity;


import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class  Member {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "member_id")
	private Long id;

	@Column(nullable = false)
	private String email;

	@Column(nullable = false)
	private String nickname;

	@Column(nullable = false)
	private String name;

	@Column(nullable = false)
	private int gender; // 0 : 남성, 1 : 여성

	@Column(nullable = false)
	private int age;

	@Column(nullable = false)
	private int readBookCount;

	@Column(nullable = false)
	private String roleName = "ROLE_USER";

	@OneToOne(fetch = FetchType.LAZY)
	@JoinColumn(name="member_image_id")
	private MemberImage memberImage;

	@Builder
	public Member(Long id, String email, String nickname, int gender, int age,
		String name, int readBookCount, String roleName, MemberImage memberImage) {
		this.id = id;
		this.email = email;
		this.nickname = nickname;
		this.gender = gender;
		this.age = age;
		this.name = name;
		this.readBookCount = readBookCount;
		this.roleName = roleName;
		this.memberImage = memberImage;
	}
}

File: ml/flask/models/Member.py
from config.dbConfig import db

class Member(db.Model):
    '''
    member_id : pk
    nickname : 닉네임
    gender : 성별
    email : 이메일
    age : 나이
    name : 이름
    read_book_count : 읽은 책 개수
    role_name : 권한
    member_image_id : 프로필 이미지 id
    '''
    __tablename__ = "member"
    
    member_id = db.Column(db.Integer,nullable = False,autoincrement=True, primary_key=True)
    nickname = db.Column(db.String(100),nullable = False)
    gender = db.Column(db.Integer,nullable = False)
    email = db.Column(db.String(200),nullable = False)
    name = db.Column(db.Integer,nullable = False)
    read_book_count = db.Column(db.Integer,nullable = False)
    role_name = db.Column(db.String(10),nullable = False)
    member_image_id = db.Column(db.Integer, nullable = False)
